sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py

```diff
@@ -250,10 +253,14 @@ def __exit__(self, exception_type, exception_value, traceback) -> None:
         if self._current_ctxt_manager:
             self._current_ctxt_manager.__exit__(exception_type, exception_value, traceback)
             self._current_ctxt_manager = None
+
         for token in self._context_tokens:
             context.detach(token)
             self._context_tokens.remove(token)
 
+        if self.previous_context:
+            context.attach(self.previous_context)
```

Comment:
without this the suppression flag never disappears after making a call in that context.